bytecomp/translcore.ml

```diff
@@ -339,12 +339,24 @@ let primitives_table = create_hashtable 57 [
   "%caml_string_get32u", Pstring_load_32(true);
   "%caml_string_get64", Pstring_load_64(false);
   "%caml_string_get64u", Pstring_load_64(true);
-  "%caml_string_set16", Pstring_set_16(false);
-  "%caml_string_set16u", Pstring_set_16(true);
-  "%caml_string_set32", Pstring_set_32(false);
-  "%caml_string_set32u", Pstring_set_32(true);
-  "%caml_string_set64", Pstring_set_64(false);
-  "%caml_string_set64u", Pstring_set_64(true);
+  "%caml_string_set16", Pbytes_set_16(false);
```

Comment:
Is there a good reason to keep the old names `%caml_string_set*`? I agree that backwards compatibility is generally good, but am just wondering.

Reply:
As @xclerc pointed out, a library such as [ocp-indent](https://github.com/OCamlPro/ocplib-endian/blob/master/src/endianString.cppo.ml#L94) will benefit from this backwards compatible change.

Reply:
Several packages were broken by a previous version of a similar change by Hongbo that turned `%string_safe_set` into `%bytes_safe_set` -- see https://github.com/ocaml/ocaml/pull/596#issuecomment-239430327 . I would be wary of changing the external-referenceable name of a primitive.